jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -661,6 +664,73 @@ def _update_stretch_histogram(self, msg={}):
         # update the n_bins since this may be a new layer
         self._histogram_nbins_changed()
 
+    @observe('is_active', 'image_color_mode_value', 'image_color_value', 'image_colormap_value',
+             'image_contrast_value', 'image_bias_value',
+             'stretch_function_value', 'stretch_vmin_value', 'stretch_vmax_value',
+             'stretch_hist_nbins', 'stretch_hist_zoom_limits')
+    @skip_if_no_updates_since_last_active()
+    def _update_stretch_histogram_colorbar(self, msg={}):
+        """Renders a colorbar on top of the histogram."""
+        if not self._viewer_is_image_viewer() or not hasattr(self, 'stretch_histogram'):
+            # don't update histogram if selected viewer is not an image viewer,
+            # or the stretch histogram hasn't been initialized:
+            return
+
+        if self.multiselect:
+            with self.stretch_histogram.hold_sync():
+                self.stretch_histogram._marks["colorbar"].x = []
+                self.stretch_histogram._marks["colorbar"].y = []
+            return
+
+        if len(self.layer.selected_obj):
+            layer = self.layer.selected_obj[0]
+        else:
+            # skip further updates if no data are available:
+            return
+
+        if isinstance(layer.layer, GroupedSubset):
+            # don't update histogram for subsets:
+            return
+
+        # Cannot use layer.state because it can be out-of-sync
+        with self.stretch_histogram.hold_sync():
+            color_mode = self.image_color_mode_value
+            interval = ManualInterval(self.stretch_vmin.value, self.stretch_vmax.value)
+            contrast_bias = ContrastBiasStretch(self.image_contrast_value, self.image_bias_value)
+            stretch = stretches.members[self.stretch_function_value]

```

Comment:
They are a bit different. You listen to `layer.state` but I cannot because the state updates lags behind so I need to grab from plugin settings.

Reply:
Ok, so can the stretch function read from plugin settings as well to be consistent?

Reply:
Isn't `self.stretch_function_value` from the plugin already?

Reply:
Yes, sorry, I meant can't _drawing_ the stretch function be done the same way as you are doing here so that the code can be consolidated?

Reply:
Maybe but I don't want to complicate it more in this PR. We're already behind schedule.

Reply:
Consolidating will make maintaining and building on top of this significantly easier though

Reply:
I am still not sure if consolidation won't introduce new bugs. `_update_stretch_curve` loops through `for layer in self.layer.selected_obj` for reason I do not understand, while `_update_stretch_histogram_colorbar` does not because I disabled it in multi-select mode. So maybe when you are looping through layers, it makes sense to access `layer.state` instead of plugin values. But I cannot do that in the colorbar because for reason I do not understand again, `layer.state` lags behind in the colorbar (i.e., if I change from Gray to Viridis, colorbar only changes to Viridis after I change it back to Gray again).

Reply:
decided to defer code consolidation and removing any multi-layer support to a follow-up effort.